cmd/client/command/common.go

```diff
@@ -132,26 +145,22 @@ func handleRequest(httpMethod string, url string, yamlBody []byte, cmd *cobra.Co
 		}
 	}
 
-	req, err := http.NewRequest(httpMethod, url, bytes.NewReader(jsonBody))
-	if err != nil {
-		ExitWithError(err)
-	}
-
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		ExitWithErrorf("%s failed: %v", cmd.Short, err)
+	tr := http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: CommandlineGlobalFlags.InsecureSkipVerify},
 	}
-	defer resp.Body.Close()
+	client := &http.Client{Transport: &tr}
+	resp, body := doRequest(httpMethod, url, jsonBody, client, cmd)
 
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		ExitWithErrorf("%s failed: %v", cmd.Short, err)
+	if resp.StatusCode == 400 && strings.Contains(string(body), "Client sent an HTTP request to an HTTPS server") {
+		fmt.Println("Warning: upgraded to HTTPS, it's better to turn on option --force-tls")
+		url = strings.ReplaceAll(url, HTTPProtocal, HTTPSProtocal)
+		resp, body = doRequest(httpMethod, url, jsonBody, client, cmd)
```

Comment:
1. L154: the error string may change, and if a nginx is deployed between egctl and easegress, the error string may also be different. Propose to only check the if the string contains `HTTP` and `HTTPS`
2. L155: I prefer a transparent upgrade and not log this message
3. L156: if the URL contains another `http`, `ReplaceAll` result in a bug (consider the code will change from time to time). and I think replace `http` with `https` is not a good pattern either. I propose to refactor `makeURL` to return an URL without the scheme, and let `handleRequest` to prepend `http` and/or `https`.